osu.Game/Graphics/UserInterface/FilteringSearchList.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using osu.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class FilteringSearchList<T> : Container where T : Drawable
+    {
+        public List<T> Items { get; }
+
+        public PropertyInfo FilterProperty
+        {
+            get { return filterProperty; }
+            private set
+            {
+                if (typeof(T).GetProperty(value.Name) != null)
+                    filterProperty = value;
+                else
+                    throw new ArgumentException($"\"{value.Name}\" property was not found in \"{typeof(T).Name}\" class");
+            }
+        }
+
+        private FlowContainer contentFlowContainer;
+        private PropertyInfo filterProperty;
+
+        public FilteringSearchList(List<T> items, PropertyInfo filterProperty)
+        {
+            Items = items;
+            FilterProperty = filterProperty;
+        }
+
+        public override void Load(BaseGame game)
+        {
+            base.Load(game);
+
+            RelativeSizeAxes = Axes.Both;
+            Children = new Drawable[]

```

Comment:
this class will be usable in zero scenarios. you're hiding the scroll and flow away from anything that would be able to do anything useful with it.

also children should be initialised int he ctor.